roles/build/tasks/build_remote.yml

```diff
@@ -0,0 +1,195 @@
+---
+- name: Build Remote - Check compiled binary
+  ansible.builtin.stat:
+    path: "{{ kaia_build_check_file_path }}"
+  register: kaia_build_check_file_stat
+
+- name: Build Remote - Check skipping compile
+  ansible.builtin.set_fact:
+    kaia_build_required: True
+  when:
+    - kaia_build_check_file_stat.stat.exists == False or 
+      kaia_build_skip_if_exists == False
+
+- name: Build Remote - Get first available host from inventory
+  ansible.builtin.set_fact:
+    remote_build_host: "{{ (groups['cn'] + groups['pn'] + groups['en'])[0] }}"
+
+- name: Build Remote - Install build dependencies
+  block:
+    - name: Update package cache
+      ansible.builtin.shell: |
+        sudo yum makecache
+      args:
+        executable: /bin/bash
+
+    - name: Install essential build tools
+      ansible.builtin.shell: |
+        sudo yum install -y git make gcc curl wget unzip which
+      args:
+        executable: /bin/bash
+
+    - name: Check if Go is installed
+      ansible.builtin.command: which go
+      register: go_check
+      ignore_errors: yes
+      changed_when: false
+
+    - name: Install Go if not present
+      block:
+        - name: Download Go
+          ansible.builtin.get_url:
+            url: "https://golang.org/dl/go1.23.0.linux-amd64.tar.gz"
+            dest: "~/go1.23.0.linux-amd64.tar.gz"
+            mode: '0644'
+          become: yes
+
+        - name: Extract Go
+          ansible.builtin.unarchive:
+            src: "~/go1.23.0.linux-amd64.tar.gz"
+            dest: "/usr/local"
+            remote_src: yes
+          become: yes
+
+        - name: Add Go to PATH
+          ansible.builtin.lineinfile:
+            path: "~/.bashrc"
+            line: 'export PATH=$PATH:/usr/local/go/bin'
+            state: present
+
+        - name: Set Go PATH for current session
+          ansible.builtin.set_fact:
+            ansible_env:
+              PATH: "{{ ansible_env.PATH }}:/usr/local/go/bin"
+      when: go_check.rc != 0
+  delegate_to: "{{ remote_build_host }}"
+  when: kaia_build_required
+
+- name: Build Remote - Create remote build directory
+  ansible.builtin.file:
+    path: "~/kaia_build"
+    state: directory
+    mode: '0755'
+  delegate_to: "{{ remote_build_host }}"
+  when:
+    - kaia_build_required
+
+- name: Build Remote - Clean remote build directory
+  ansible.builtin.file:
+    state: absent
+    path: "~/kaia_build"
+  delegate_to: "{{ remote_build_host }}"
+  when:
+    - kaia_build_required
+
+- name: Build Remote - Git checkout
+  ansible.builtin.git:
+    repo: "{{ kaia_build_remote_git_url }}"
```

Comment:
Would you support multi git repo?
* https://github.com/kaiachain/kaiaspray/pull/62/files